runtime/dart_isolate.cc

```diff
@@ -0,0 +1,615 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/runtime/dart_isolate.h"
+
+#include <tuple>
+
+#include "flutter/common/threads.h"
+#include "flutter/lib/io/dart_io.h"
+#include "flutter/lib/ui/dart_runtime_hooks.h"
+#include "flutter/lib/ui/dart_ui.h"
+#include "flutter/runtime/dart_service_isolate.h"
+#include "flutter/runtime/dart_vm.h"
+#include "lib/fxl/files/path.h"
+#include "lib/tonic/converter/dart_converter.h"
+#include "lib/tonic/dart_class_library.h"
+#include "lib/tonic/dart_class_provider.h"
+#include "lib/tonic/dart_message_handler.h"
+#include "lib/tonic/dart_state.h"
+#include "lib/tonic/dart_sticky_error.h"
+#include "lib/tonic/file_loader/file_loader.h"
+#include "lib/tonic/scopes/dart_api_scope.h"
+#include "lib/tonic/scopes/dart_isolate_scope.h"
+#include "third_party/dart/runtime/include/dart_tools_api.h"
+
+namespace blink {
+
+fxl::WeakPtr<DartIsolate> DartIsolate::CreateRootIsolate(
+    const DartVM* vm,
+    Threads threads,
+    std::unique_ptr<Window> window,
+    std::string advisory_script_uri,
+    std::string advisory_script_entrypoint) {
+  Dart_Isolate vm_isolate = nullptr;
+  fxl::WeakPtr<DartIsolate> embedder_isolate;
+
+  char* error = nullptr;
+
+  // Since this is the root isolate, we fake a parent embedder data object. We
+  // cannot use unique_ptr here because the destructor is private (since the
+  // isolate lifecycle is entirely managed by the VM).
+  DartIsolate* root_embedder_data = new DartIsolate(
+      vm, std::move(threads), advisory_script_uri, advisory_script_entrypoint);
```

Comment:
Why use std::move here?  blink::Threads doesn't have a move constructor and its state isn't being migrated.

Reply:
I'll get rid of the move.